### PR TITLE
Add ipv6 support.

### LINKED ISSUE #380: in-cluster config builder does not work properly in an ipv6 cluster
https://github.com/kubernetes-client/javascript/blob/d262aff9acc43efda602142a587baa0183c4970a/src/config.ts#L204

assumes a DNS name or ipv4 address... it does not work correctly with an ipv6 address (the ipv6 host must be wrapped in brackets)

### PR DESCRIPTION
**Pull request by brendandburns**
Fixes #380 

cc @liggett @thockin

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -197,11 +197,17 @@ export class KubeConfig {
             scheme = 'http';
         }
 
+        // Wrap raw IPv6 addresses in brackets.
+        let serverHost = host;
+        if (host && host.indexOf(':') != -1) {
```
Comment (drubin) [inline src/config.ts]: There are methods to check if its an ipv6 Adress.

File: src/config.ts [hunk 2]
```diff
@@ -197,11 +197,17 @@ export class KubeConfig {
...
+            serverHost = `[${host}]`;
+        }
+
         this.clusters = [
             {
                 name: clusterName,
                 caFile: `${pathPrefix}${Config.SERVICEACCOUNT_CA_PATH}`,
-                server: `${scheme}://${host}:${port}`,
+                server: `${scheme}://${serverHost}:${port}`,
```
Comment (drubin) [inline src/config.ts]: Maybe we should also be building up the address with Uri helpers? Maybe they can also handle this natively?

File: src/config_test.ts [hunk 3]
```diff
@@ -1073,6 +1073,32 @@ describe('KubeConfig', () => {
...
+            mockfs.restore();
+            delete process.env.KUBERNETES_SERVICE_HOST;
+            delete process.env.KUBERNETES_SERVICE_PORT;
+
+            const cluster = kc.getCurrentCluster();
+            expect(cluster).to.not.be.null;
+            if (!cluster) {
+                return;
+            }
+            expect(cluster.server).to.equal('http://[::1234:5678]:80');
```
Comment (drubin) [inline src/config_test.ts]: Thanks for adding a test!

File: src/config.ts (hunk 1 above)
Comment (brendandburns) [reply src/config.ts]: done.

File: src/config.ts (hunk 2 above)
Comment (brendandburns) [reply src/config.ts]: I couldn't tell from docs if they handled this or not, so I left it as is.

**Comment by brendandburns** [comment]
Comments addressed, style fixed, please take another look.

Thanks!

**Pull request merged**